### PR TITLE
Show a warning dialog if the text of the clicked link does not match

### PR DESCRIPTION
**Pull request by onurays**
Screenshot_1587652730

### CODE REVIEW

**Comment by nadonomy** [comment]
Some simple visual requests here:

https://www.figma.com/file/qybYwPSR9HpsneLl29R8wX/Non-matching-links?node-id=1%3A43

**Comment by BillCarsonFr** [comment]
Fixes #922

File: vector/src/main/java/im/vector/riotx/core/utils/EvenBetterLinkMovementMethod.kt [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+package im.vector.riotx.core.utils
+
+import android.text.Spanned
+import android.text.style.ClickableSpan
+import android.text.style.URLSpan
+import android.widget.TextView
+import me.saket.bettermovementmethod.BetterLinkMovementMethod
+
+class EvenBetterLinkMovementMethod(private val onLinkClickListener: OnLinkClickListener? = null) : BetterLinkMovementMethod() {
```
Comment (BillCarsonFr) [inline vector/src/main/java/im/vector/riotx/core/utils/EvenBetterLinkMovementMethod.kt:25]: 😆

File: vector/src/main/res/values/strings_riotX.xml [hunk 2]
```diff
@@ -21,7 +21,10 @@
 
 
     <!-- BEGIN Strings added by Onuray -->
-
+    <string name="external_link_confirmation_title">Double-check this link</string>
+    <string name="external_link_confirmation_message">The link %1$s is taking you to another site:\n%2$s. Are you sure you want to continue?</string>
+    <string name="external_link_confirmation_negative_button">Continue</string>
```
Comment (BillCarsonFr) [inline vector/src/main/res/values/strings_riotX.xml:26]: there are generic cancel / continue strings already

File: vector/src/main/java/im/vector/riotx/features/home/room/detail/RoomDetailFragment.kt [hunk 3]
```diff
@@ -947,8 +951,25 @@ class RoomDetailFragment @Inject constructor(
                 .observeOn(AndroidSchedulers.mainThread())
                 .subscribe { managed ->
                     if (!managed) {
-                        // Open in external browser, in a new Tab
-                        openUrlInExternalBrowser(requireContext(), url)
+                        if (title.isValidUrl() && URL(title).host != URL(url).host) {
```
[the remark this answers is not in the document]
Comment (onurays) [reply vector/src/main/java/im/vector/riotx/features/home/room/detail/RoomDetailFragment.kt:954]: It is checked at createLinkMovementMethod, but I will add the condition here too :)

File: vector/src/main/java/im/vector/riotx/features/home/room/detail/timeline/tools/EventRenderingTools.kt [hunk 4]
```diff
@@ -54,14 +57,13 @@ fun CharSequence.linkify(callback: TimelineEventController.UrlClickCallback?): C
 
 // Better link movement methods fixes the issue when
 // long pressing to open the context menu on a TextView also triggers an autoLink click.
-fun createLinkMovementMethod(urlClickCallback: TimelineEventController.UrlClickCallback?): BetterLinkMovementMethod {
-    return BetterLinkMovementMethod.newInstance()
+fun createLinkMovementMethod(urlClickCallback: TimelineEventController.UrlClickCallback?): EvenBetterLinkMovementMethod {
+    return EvenBetterLinkMovementMethod(object : EvenBetterLinkMovementMethod.OnLinkClickListener {
+        override fun onLinkClicked(textView: TextView, span: ClickableSpan, url: String, actualText: String): Boolean {
+            return url.isValidUrl() && urlClickCallback?.onUrlClicked(url, actualText) == true
```
[the remark this answers is not in the document]
Comment (onurays) [reply vector/src/main/java/im/vector/riotx/features/home/room/detail/timeline/tools/EventRenderingTools.kt:63]: Moved to the fragment.

**Pull request merged**